stdlib/public/core/Integers.swift

```diff
@@ -1315,6 +1315,161 @@ extension BinaryInteger {
     return self.magnitude % other.magnitude == 0
   }
 
+  /// Returns `true` if this value is a power of the given base, and `false`
+  /// otherwise.
+  ///
+  /// For two integers *a* and *b*, *a* is a power of *b* if *a* is equal to
+  /// any repeated multiplication of *b*. For example:
+  ///
+  /// * `16.isPower(of: 2)` is `true`, because `16 == (2 * 2 * 2 * 2)`
+  /// * `(-27).isPower(of: -3)` is `true`, because `-27 == (-3 * -3 * -3)`
+  ///
+  /// Note that one is a power of anything, because any number to the zero
+  /// power is considered an empty product, which equals one.
+  ///
+  /// For the corner case where base is zero, `x.isPower(of: 0)` is `true` if
+  /// `x` is either zero or one, and `false` otherwise.
+  ///
+  /// - Parameter base: The base value to test.
+  @_alwaysEmitIntoClient
+  public func isPower(of base: Self) -> Bool {
+    // Fast path when base is one of the common cases.
+    if base == 2 { return self._isPowerOfTwo }
+    if base == 10 { return self._isPowerOfTen }
+    if base._isPowerOfTwo { return self._isPowerOf(powerOfTwo: base) }
```

Comment:
Should the `_isPowerOf(powerOfTwo:)` function be removed?

```suggestion
    if base._isPowerOfTwo {
      return self._isPowerOfTwo &&
        self.trailingZeroBitCount.isMultiple(of: base.trailingZeroBitCount)
    }
```

Reply:
I don't think it necessary, though it is good to reduce the number of APIs. IMO, inlining the code seems to make the public function's implementation less concise and less consistent across branches (calls vs inlining code).

Anyway, I think either way is fine. You call.